### PR TITLE
#316: Adds support for http/https APIs for node >= 10

### PR DESCRIPTION
**Pull request by kennethaasan**

### CODE REVIEW

File: lib/instrumentation/core/http-outbound.js [hunk 1]
```diff
@@ -125,11 +127,12 @@ module.exports = function instrumentOutbound(agent, opts, makeRequest) {
     segment.start()
     const request = makeRequest(opts)
     const parsed = urltils.scrubAndParseParameters(request.path)
-    const proto = parsed.protocol || opts.protocol || 'http:'
+    const proto = parsed.protocol || _url.protocol || 'http:'
     segment.name += parsed.path
-    Object.defineProperty(request, '__NR_segment', {
-      value: segment
-    })
+    // Object.defineProperty(request, '__NR_segment', {
```
Comment (kennethaasan) [inline lib/instrumentation/core/http-outbound.js]: With this set, it throws `TypeError: Cannot redefine property: __NR_segment`

File: test/unit/instrumentation/http/outbound.test.js [hunk 2]
```diff
@@ -305,7 +320,7 @@ describe('should add data from cat header to segment', function() {
     helper.runInTransaction(agent, function() {
       addSegment()
       http.get({host : 'localhost', port : 4123}, function(res) {
-        var segment = agent.tracer.getTransaction().trace.root.children[0]
+        var segment = agent.tracer.getTransaction().trace.root.children[0].children[0]
```
Comment (kennethaasan) [inline test/unit/instrumentation/http/outbound.test.js]: Not sure why this happens?

File: test/unit/instrumentation/http/outbound.test.js [hunk 3]
```diff
@@ -322,7 +337,7 @@ describe('should add data from cat header to segment', function() {
...
```
Comment (kennethaasan) [inline test/unit/instrumentation/http/outbound.test.js]: Not sure why this happens?

File: lib/instrumentation/core/http.js [hunk 4]
```diff
@@ -537,22 +561,6 @@ module.exports = function initialize(agent, http, moduleName) {
...
-    // This means we need to skip instrumenting to avoid double
-    // instrumenting external requests.
-    if (!IS_HTTPS && psemver.satisfies('>=8')) {
-      shimmer.wrapMethod(
-        http,
-        'http',
-        'get',
-        wrapRequest.bind(null, agent)
-      )
-    }
```
Comment (kennethaasan) [inline lib/instrumentation/core/http.js]: This needs to be addressed. This creates double calls to _wrapRequest_

File: test/unit/instrumentation/http/outbound.test.js (hunk 2 above)
Comment (kennethaasan) [reply test/unit/instrumentation/http/outbound.test.js]: Fixed

File: test/unit/instrumentation/http/outbound.test.js (hunk 3 above)
Comment (kennethaasan) [reply test/unit/instrumentation/http/outbound.test.js]: Fixed

**Comment by nijotz** [comment]
Hi Kenneth, I just started working on this issue, looking to fix and release soon. I'm curious if you still had that branch and wanted to PR?

**Comment by nijotz** [comment]
Oh nevermind, I just found your other pull request.

**Pull request closed without merging**